pkg/kademlia/kademlia.go

```diff
@@ -637,6 +884,53 @@ func (k *Kad) neighborhoodDepth() uint8 {
 	return k.depth
 }
 
+// IsBalanced returns if Kademlia is balanced to bin.
+func (k *Kad) IsBalanced(bin uint8) bool {
+	k.depthMu.RLock()
+	defer k.depthMu.RUnlock()
+
+	if int(bin) > len(k.commonBinPrefixes) {
+		return false
+	}
+
+	binCheckBitLen := int(bin) + k.bitSuffixLength
```

Comment:
I think this should be int(bin) + k.bitSuffixLength + 1 because of flipped bit in common addresses